### PR TITLE
Integration spec, Fix n+1 problems

### PR DESCRIPTION
**Pull request by ridabensalem**
in this milestone of blog app we have added the following requirements : 
Before Fixing N+1 Problems
Before_n+1
After  Fixing N+1 Problems
After_n+1

- Use Capybara to write integration tests for each view in your project.
- Test the following features :

- User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.

- user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.

- Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Rida, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Unable to attain desired results on test execution due to database errors. Kindly ensure to resolve the same and resubmit for review:

image

Also Linters are not showing on the Pull Request. Kindly ensure that Github actions is properly configured Linters

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    end
+  end
+
+  scenario 'I can see the number of posts each user has written' do
+    visit users_path
+    User.all.each do |user|
+      expect(page).to have_content("Number of posts: #{user.posts_counter}")
+    end
+  end
+end
```
Comment (mckent05) [inline spec/views/user_index_spec.rb]: - Good job done so far, but according to the project requirements, you should also add a test that redirects to the user's show page when you click on a user. Please fix this 🙏

File: spec/views/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    scenario 'When I click a users post, it redirects me to that posts show page.' do
+      visit user_path
+      click_link 'Post'
+      expect(page).to have_content(post.title)
+    end
+    scenario 'I can see the users first 3 posts'
+    visit user_path
+    expect(page).to have_content(user.posts.first(3))
+  end
+end
```
Comment (mckent05) [inline spec/views/user_show_spec.rb]: - Good job done so far, but according to the project requirements, your `user#show` page is missing two tests, please add these tests:
1. Tests that assert you can see a button that lets me view all of a user's posts.
2. Tests that redirect to the user's post's index page when you click on a **see all posts**.

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @ridabensalem, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding integration tests to your view files 👏 👏 

## Required Changes ♻️
- Good jib done so far, but according to the project requirements, this section of your gemfile should look like this 🙏:
```ruby
group :development, :test do
  # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-
  gem 'debug', platforms: %i[mri mingw x64_mingw]
  gem 'capybara'
end
```
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @ridabensalem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] Tests are passing ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**